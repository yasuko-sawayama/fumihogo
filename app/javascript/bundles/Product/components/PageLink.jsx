import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';
import { join } from '../../shared/utils/join';
import styled from 'styled-components';

const StyledLink = styled.li`
  .active {
    color: #fff;
    background-color: #eb6864;
  }
`;

const PageLink = ({ id, title, url, }) => (
  <StyledLink>
    <NavLink activeClassName="active" to={join(url, `/pages/${id}`)}>
      {id}. {title}
    </NavLink>
  </StyledLink>
);

PageLink.propTypes = {
  id: PropTypes.number.isRequired,
  url: PropTypes.string.isRequired,
  title: PropTypes.string,
};

export default PageLink;
