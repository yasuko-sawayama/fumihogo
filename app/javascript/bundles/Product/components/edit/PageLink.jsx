import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';
import { join } from '../../../shared/utils/join';
import styled from 'styled-components';

const StyledLink = styled(NavLink).attrs('active')`
&.active {
    color: #fff;
    background-color: #EB6864;
}
`;

const PageLink = ({
  id, api, position, title, url,
}) => (
  <li>
    <StyledLink activeClassName="active" to={join(url, `/pages/${id}`)}>
      {position}.{' '}
      {title}
    </StyledLink>
  </li>
);

PageLink.propTypes = {
  id: PropTypes.number.isRequired,
  position: PropTypes.number.isRequired,
  url: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
};

export default PageLink;
