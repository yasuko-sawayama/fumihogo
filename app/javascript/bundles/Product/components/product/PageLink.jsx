import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';
import { join } from '../../../shared/utils/join';

const PageLink = ({ id, api, order, title, url }) => (
  <li>
    <NavLink activeClassName="active" to={join(url, `/pages/${id}`)}>{title}</NavLink>
  </li>
);

PageLink.propTypes = {
  id: PropTypes.number.isRequired,
  order: PropTypes.number.isRequired,
  url: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
};

export default PageLink;
