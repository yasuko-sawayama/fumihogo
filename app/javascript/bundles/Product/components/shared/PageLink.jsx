import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { NavLink } from 'react-router-dom';
import { join } from '../../../shared/utils/join';

const StyledLink = styled.li`
  .active {
    color: #fff;
    background-color: #eb6864;
  }
`;

const PageLink = ({ id, title, url }) => (
  <StyledLink>
    <NavLink activeClassName="active" to={join(url, `/pages/${id}`)}>
      {id}. {title}
    </NavLink>
  </StyledLink>
);

PageLink.defaultProps = {
  title: ''
};

PageLink.propTypes = {
  id: PropTypes.number.isRequired,
  url: PropTypes.string.isRequired,
  title: PropTypes.string
};

export default PageLink;
