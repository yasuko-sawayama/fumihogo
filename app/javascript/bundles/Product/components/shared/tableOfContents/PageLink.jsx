import React from "react";
import PropTypes from "prop-types";
import { NavLink } from "react-router-dom";
import styled from "styled-components";

import { pageUrl } from "../../../libs/utils";

const StyledLink = styled(NavLink).attrs("active")`
  &.active {
    color: #fff;
    background-color: #eb6864;
  }
`;

const PageLink = ({ api, position, title, productId }) => (
  <li>
    <StyledLink activeClassName="active" to={pageUrl(position, productId)}>
      {position}. {title}
    </StyledLink>
  </li>
);

PageLink.propTypes = {
  position: PropTypes.number.isRequired,
  productId: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired
};

export default PageLink;
