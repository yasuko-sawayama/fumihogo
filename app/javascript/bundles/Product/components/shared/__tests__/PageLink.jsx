import React from "react";
import { NavLink } from "react-router-dom";
import { MemoryRouter } from "react-router";
import { mount, shallow } from "enzyme";

import PageLink from "../PageLink";

describe("PageLink", () => {
  it("正しいURLが生成できていること", () => {
    const url = "http://example.com/products/1/";
    const tree = shallow(<PageLink id={1} url={url} />);
    const button = tree.find(NavLink);
    expect(button.props().to).toBe("http://example.com/products/1/pages/1");
  });

  it("タイトルが表示できること", () => {
    const url = "http://example.com/products/1/";
    const title = "ページタイトル";
    const node = (
      <MemoryRouter>
        <PageLink id={1} url={url} title={title} />
      </MemoryRouter>
    );

    const tree = mount(node);

    expect(tree).toHaveText("1. ページタイトル");
  });
});
