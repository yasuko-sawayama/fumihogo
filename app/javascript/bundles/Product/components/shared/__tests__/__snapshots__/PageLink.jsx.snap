// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageLink render 1`] = `
<li>
  <PageLink__StyledLink
    activeClassName="active"
    to="/1/pages/1"
  >
    1
    . 
    タイトル
  </PageLink__StyledLink>
</li>
`;
